### PR TITLE
[Android] Fix Button IconSize

### LINKED ISSUE #12054: [Android] A big image overflow the Button
### Description

A big image is not resized or adapted and can overflow the Button.

image

```
<Button
    HeightRequest="60"
    ImageSource="dotnet_bot.png"
    Text="Default" />
<Button
    HeightRequest="60"
    ImageSource="dotnet_bot.png"
    Text="Left - 10"
    ContentLayout="Left, 10" />
<Button
    HeightRequest="60"
    ImageSource="dotnet_bot.png"
    Text="Top - 10"
    ContentLayout="Top, 10" />
```

### Steps to Reproduce

1. Download or sync https://github.com/jsuarezruiz/dotnet-maui-testing-playground
2. Launch the App and navigate to the Button test cases.
3. The B9 test case validating the ContentLayout and ImageSource properties is failing.

### Link to public reproduction project repository

https://github.com/jsuarezruiz/dotnet-maui-testing-playground

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 13

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change

Fix Button IconSize on Android.

Before

After
Captura de pantalla 2023-07-12 a las 14 14 57

To validate the changes, launch the .NET MAUI Gallery and navigate to the Button samples. Check the added samples using the huge dotnet bot png.

Captura de pantalla 2023-07-17 a las 9 24 32
Captura de pantalla 2023-07-17 a las 11 34 18

### Issues Fixed

Fixes #12054

### CODE REVIEW

**Comment by jfversluis** [comment]
From the images it looks like the aspect is not preserved (it's stretched or squashed) is that something we should do?

File: src/Controls/src/Core/Button/Button.Android.cs [hunk 1]
```diff
@@ -1,13 +1,35 @@
...
+		private protected override void OnHandlerChangedCore()
+		{
+			base.OnHandlerChangedCore();
+
+			if (Handler != null)
+			{
+				if (Handler is ButtonHandler buttonHandler && buttonHandler.PlatformView is MaterialButton materialButton)
+				{
+					_materialButton = materialButton;
+					_materialButton.LayoutChange += OnButtonLayoutChange;
```
Comment (PureWeen) [inline src/Controls/src/Core/Button/Button.Android.cs:20]: This is going to cause memory leaks. 

Can we trigger this logic from the handler without events?

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Button/Button.Android.cs (hunk 1 above)
Comment (jsuarezruiz) [reply src/Controls/src/Core/Button/Button.Android.cs]: Reviewed and added a way to always unsubscribe with test to validate that the Button does not leak.
Captura de pantalla 2023-07-17 a las 11 34 18

**Review by jfversluis — DISMISSED** [review]

File: src/Controls/src/Core/Button/Button.Android.cs [hunk 2]
```diff
@@ -1,13 +1,41 @@
...
 
 namespace Microsoft.Maui.Controls
 {
 	public partial class Button
 	{
+		MaterialButton _materialButton;
+
+		private protected override void OnHandlerChangedCore()
+		{
+			base.OnHandlerChangedCore();
```
Comment (PureWeen) [inline src/Controls/src/Core/Button/Button.Android.cs:13]: This isn't going to always fire. Can we make this work without needing to subscribe to LayoutChange from the controls code?

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Button/Button.Android.cs (hunk 2 above)
Comment (jsuarezruiz) [reply src/Controls/src/Core/Button/Button.Android.cs:13]: We are also unsubscribing when the parent Window is null. I have added a test to validate leaks. What would be a possible scenario for testing?

**Comment by PureWeen** [comment]
Fixed by 
https://github.com/dotnet/maui/pull/19834

**Pull request closed without merging**